Add SchemaRegistry utils and schema_id checking in poller

helgeholm opened the pull request:
Draft - needs major version bump

helgeholm committed:
Add SchemaRegistry utils and schema_id checking in poller

theothornhill reviewed (commented):
Nice start!

What do you think about also supporting this on a producer init?

And also to fetch and save latest schema in a directory for zig-avro to find and generate structs from

helgeholm commented on lib/consumer.zig:
```diff
@@ -91,7 +91,13 @@ pub fn Consumer(comptime T: type) type {
 
             if (message.payload) |p| {
-                _ = try avro.Reader.read(T, &self.current_message.payload.?, str(p, message.len)[5..]);
+                const inbuf = str(p, message.len);
```
Currently, if we read a topic with multiple yet compatible IDs, the application corrupts data.

This change does 1 call to the SR and asserts that we are not trying to parse data we can't handle.

Agree on where we want to go wrt compatibility, but that's *a lot* of both thinking and tooling and out of the intended scope for this PR.

helgeholm committed:
Let schema_id be Consumer config

Since we're already "locking" the expected schema, it makes more sense to
also set exactly one schema_id.
